29th_first_semina/app/src/main/java/com/example/a29th_first_semina/SignInActivity.kt

```diff
@@ -0,0 +1,62 @@
+package com.example.a29th_first_semina
+
+import android.content.Intent
+import androidx.appcompat.app.AppCompatActivity
+import android.os.Bundle
+import android.util.Log
+import android.widget.Toast
+import com.example.a29th_first_semina.databinding.ActivitySigninBinding
+
+class SignInActivity : AppCompatActivity() {
+    private lateinit var binding: ActivitySigninBinding
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        binding = ActivitySigninBinding.inflate(layoutInflater)
+        setContentView(binding.root)
+
+        initButtonClickEvent()
+
+    }
+
+    private fun initButtonClickEvent() {
+
+        clickLoginEvent()
+        clickSignUpEvent()
+    }
+
+    private fun moveHomeActivity() {
+        val intent = Intent(this, HomeActivity::class.java)
+        startActivity(intent)
+    }
+
+    private fun moveSignUpActivity() {
+        val signupintent = Intent(this, SignUpActivity::class.java)
+        startActivity(signupintent)
+
+    }
+
+    private fun clickLoginEvent() {
+
+        binding.apply {
+            btnLogin.setOnClickListener {
+                val userId: String = etId.text.toString()
+                val userPwd: String = etPwd.text.toString()
+                if (userId.isEmpty() || userPwd.isEmpty()) {
+                    Toast.makeText(this@SignInActivity, "로그인 실패", Toast.LENGTH_SHORT).show()
+                } else {
+                    moveHomeActivity()
+                }
+
+            }
+
+        }
```

Comment:
```suggestion
        with(binding) {
            btnLogin.setOnClickListener {
                val userId: String = etId.text.toString()
                val userPwd: String = etPwd.text.toString()
                if (userId.isEmpty() || userPwd.isEmpty()) {
                    Toast.makeText(this@SignInActivity, "로그인 실패", Toast.LENGTH_SHORT).show()
                } else {
                    moveHomeActivity()
                }
            }
        }
```

1. 일반적으로 apply 함수는 객체 자신을 다시 반환 하려는 경우에 사용합니다. 지금은 어떤 값을 반환받을 필요가 없기에 with를 사용하는 것이 조금 더 적절한 use case일 것 같습니다(동작은 같습니다만, 역할이 다릅니다)
2. pwd와 같은 축약어는 사용하지 않는 것이 가독성 증진에 기여합니다. userPwd보단 userPassword가 한눈에 더 이해가 쉽게 되지 않을까요?
3. etId.text.isEmpty()로만 해도 Edittext 내부의 text가 비었는 지 안 비었는 지 확인 가능합니다.
4. moveToHomeActivity()와 같은 경우 한 눈에 알아보기 쉽게 작명이 되어있습니다. 정말 좋네요

Reply:
오 안그래도 apply와 with 차이점 좀 공부하려고 했는데 감쟈합니다